proxy.js

```diff
@@ -1024,7 +1024,6 @@ function activateHTTP() {
 				for (let minerID in activeWorkers[workerID]){
                 			if (!activeWorkers[workerID].hasOwnProperty(minerID)) continue;
 					let miner = activeWorkers[workerID][minerID];
-					if (!miner) continue;
 					if (!miner.identifier) {
```

Comment:
I actually meant this check). The check that you are removed can be OK if you have evidence it was really needed.

Reply:
In fact, it is "if (!miner.identifier)" what I tried to fix my proxy. I got undefined identifier error, proxy crashed. However, even with that if check, it still happened after that check. So it could be non sense, I think it's still useful if identifier == null.
For undefined identifier error, it happened when pool failed to get tempale. Can you think about any ideas to avoid undefined identifier ?

Reply:
I flushed the log. Let's wait for next fail to get template and I'll paste the log here. Keep temporarily this. If it won't happen again, I'll remove it.

Reply:
Here we go. How can I fix this issue (your codes, if check removed) ?
1|proxy    | 2018-04-23 22:35 +00:00: TypeError: Cannot read property 'identifier' of undefined
1|proxy    |     at Server.http.createServer (/home/pooldaemon/xmr-node-proxy/proxy.js:1030:24)
1|proxy    |     at emitTwo (events.js:126:13)
1|proxy    |     at Server.emit (events.js:214:7)
1|proxy    |     at parserOnIncoming (_http_server.js:602:12)
1|proxy    |     at HTTPParser.parserOnHeadersComplete (_http_common.js:116:23)
![identifiererror](https://user-images.githubusercontent.com/29669374/39156916-88ebc8c8-4782-11e8-84b4-e7a63a484390.png)

Reply:
Under my debug, even with that if check, it still occurs. That is OUFFF. I can’t understand.

Reply:
OK, to avoid this error you should keep "if (!miner) continue;" (because it is miner that can be unauthorized here) check and remove "if (!miner.identifier)" check. 

Can you please fix that in your patch so I can merge it? (I want to keep your name to credit you as author of these changes)